test/test_esapi.py

```diff
@@ -2843,6 +2843,30 @@ def test_GetSessionAuditDigest(self):
                 signHandle, session, b"1234", session3=12.723
             )
 
+    def test_PP_Commands(self):
+        with self.assertRaises(TSS2_Exception) as e:
+            self.ectx.PP_Commands(TPML_CC(), TPML_CC(), session1=ESYS_TR.PASSWORD)
+        self.assertEqual(e.exception.error, TPM2_RC.PP)
+        self.assertEqual(e.exception.session, 1)
+
+        with self.assertRaises(TypeError):
+            self.ectx.PP_Commands(b"bad setList", TPML_CC(), session1=ESYS_TR.PASSWORD)
+
+        with self.assertRaises(TypeError):
+            self.ectx.PP_Commands(TPML_CC(), None, session1=ESYS_TR.PASSWORD)
+
+        with self.assertRaises(TypeError):
+            self.ectx.PP_Commands(TPML_CC(), None, session1=b"0xF1F1")
+
+    def test_SetAlgorithmSet(self):
+        self.ectx.SetAlgorithmSet(0)
+
+        with self.assertRaises(TypeError):
+            self.ectx.SetAlgorithmSet([1, 2, 3])
+
+        with self.assertRaises(TypeError):
+            self.ectx.SetAlgorithmSet(session2=set(3, 2, 1))
```

Comment:
Test each and every input as tedious as it is:
1. This is to ensure we get all the TypeError behavior right in the API.
2. This also helps bump test coverage, because we're testing the exception paths.

Look at the tests, like test_CreatePrimary:
  - https://github.com/tpm2-software/tpm2-pytss/blob/master/test/test_esapi.py#L82

for an example.

Reply:
Sure, it should be done now and I'll remember that for the next set of ESAPI tests